testtools/content.py

```diff
@@ -163,61 +161,52 @@ def __init__(self, stack_lines, prefix_content="", postfix_content=""):
     def _stack_lines_to_unicode(self, stack_lines):
         """Converts a list of pre-processed stack lines into a unicode string.
         """
+        msg_lines = traceback.format_list(stack_lines)
+        return _u('').join(msg_lines)
 
-        # testtools customization. When str is unicode (e.g. IronPython,
-        # Python 3), traceback.format_exception returns unicode. For Python 2,
-        # it returns bytes. We need to guarantee unicode.
-        if str_is_unicode:
-            format_stack_lines = traceback.format_list
-        else:
-            format_stack_lines = _format_stack_list
 
-        msg_lines = format_stack_lines(stack_lines)
-
-        return ''.join(msg_lines)
-
-
-def TracebackContent(err, test):
+class TracebackContent(Content):
     """Content object for tracebacks.
 
     This adapts an exc_info tuple to the 'Content' interface.
     'text/x-traceback;language=python' is used for the mime type, in order to
     provide room for other languages to format their tracebacks differently.
     """
-    if err is None:
-        raise ValueError("err may not be None")
-
-    exctype, value, tb = err
-    # Skip test runner traceback levels
-    if StackLinesContent.HIDE_INTERNAL_STACK:
-        while tb and '__unittest' in tb.tb_frame.f_globals:
-            tb = tb.tb_next
-
-    # testtools customization. When str is unicode (e.g. IronPython,
-    # Python 3), traceback.format_exception_only returns unicode. For Python 2,
-    # it returns bytes. We need to guarantee unicode.
-    if str_is_unicode:
-        format_exception_only = traceback.format_exception_only
-    else:
-        format_exception_only = _format_exception_only
 
-    limit = None
-    # Disabled due to https://bugs.launchpad.net/testtools/+bug/1188420
-    if (False
-        and StackLinesContent.HIDE_INTERNAL_STACK
-        and test.failureException
-        and isinstance(value, test.failureException)):
-        # Skip assert*() traceback levels
-        limit = 0
-        while tb and not self._is_relevant_tb_level(tb):
-            limit += 1
-            tb = tb.tb_next
+    def __init__(self, err, test, capture_locals=False):
+        """Create a TracebackContent for ``err``.
 
-    prefix = _TB_HEADER
-    stack_lines = traceback.extract_tb(tb, limit)
-    postfix = ''.join(format_exception_only(exctype, value))
-
-    return StackLinesContent(stack_lines, prefix, postfix)
+        :param err: An exc_info error tuple.
+        :param test: A test object used to obtain failureException.
+        :param capture_locals: If true, show locals in the traceback.
+        """
+        if err is None:
+            raise ValueError("err may not be None")
+
+        exctype, value, tb = err
```

Comment:
Its what the prior code did. The indent has shifted is all. Valid point but I'm meh either way today :)